### ISSUE TITLE
Failed to check registry.npmjs.org for updates: "Cannot read properties of undefined (reading 'length')

### ISSUE THREAD

**Issue by anbie** [oznu/homebridge-config-ui-x#2077]
### Current Situation

I'm starting to get these in my log every few 20 min. 
> Failed to check registry.npmjs.org for updates: "Cannot read properties of undefined (reading 'length')"

I have not found a way to cure this. Things like npm clear cache did not help.

### Logs

```markdown
[24/06/2024, 14:38:49] [Homebridge UI] [homebridge-config-ui-x] Failed to check registry.npmjs.org for updates: "Cannot read properties of undefined (reading 'length')" - see https://homebridge.io/w/JJSz6 for help.
[24/06/2024, 14:38:49] [Homebridge UI] [homebridge-config-ui-x] Failed to check registry.npmjs.org for updates: "Cannot read properties of undefined (reading 'length')" - see https://homebridge.io/w/JJSz6 for help.
```

### Environment

- OS: Linux homebridge 6.1.21-v8+
- Software:
- Node: v20.15.0
- npm: 10.7.0

### Process Supervisor

hb-service

**Comment by NorthernMan54** [comment]
Over the weekend something changed on the npmjs.com site, and they have stopped supplying a documented field 'maintainers'.

- NPM always add maintainers to package, now stopped.
- Homebridge had a bug that checked maintainers.lenght before checking if it exists at all. It was fixes. homebridge/homebridge#2069 
- Homebridge check maintainers from latest stable release of plugin, and first of maintainers you see on plugin card next to heart icon.
- So if plugin doesnt have any new release since 2-3 days (since npm does not add maintainers), everything works.

To summarize:

- NPM error is that it does not add maintainers (since couple days).
- The Homebridge UI error was that it checked how many maintainers, without checking if they were at all exist. And this suspended all npm functions, also responsible for checking the version, etc. If it weren't for that, the NPM error would only cause that there would be no author on the plugin card (only a heart icon and a @ sign because it is hardcoded).

To resolve:

- Manually update homebridge UI to the latest version 4.56.3

![Screenshot 2024-06-24 at 09 05 42](https://github.com/homebridge/homebridge-config-ui-x/assets/17454593/21f34241-c2b2-483e-a439-987da950eeab)

**Comment by NorthernMan54** [comment]
Pls add your upvotes to this issue on the npm side https://github.com/orgs/community/discussions/129220

https://github.com/orgs/community/discussions

**Comment by mkz212** [comment]
> Over the weekend something changed on the npmjs.com site, and they have stopped supplying a documented field 'maintainers'.
> 
> * NPM always add maintainers to package, now stopped.
> * Homebridge had a bug that checked maintainers.lenght before checking if it exists at all. It was fixes. [Homebridge Wont Wake homebridge#2069](https://github.com/homebridge/homebridge/issues/2069)
> * Homebridge check maintainers from latest stable release of plugin, and first of maintainers you see on plugin card next to heart icon.
> * So if plugin doesnt have any new release since 2-3 days (since npm does not add maintainers), everything works.
> 
> To summarize:
> 
> * NPM error is that it does not add maintainers (since couple days).
> * The Homebridge UI error was that it checked how many maintainers, without checking if they were at all exist. And this suspended all npm functions, also responsible for checking the version, etc. If it weren't for that, the NPM error would only cause that there would be no author on the plugin card (only a heart icon and a @ sign because it is hardcoded).
> 
> To resolve:
> 
> * Manually update homebridge UI to the latest version 4.56.3
> 
> ![Screenshot 2024-06-24 at 09 05 42](https://private-user-images.githubusercontent.com/17454593/342236674-21f34241-c2b2-483e-a439-987da950eeab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkyNDU5MDcsIm5iZiI6MTcxOTI0NTYwNywicGF0aCI6Ii8xNzQ1NDU5My8zNDIyMzY2NzQtMjFmMzQyNDEtYzJiMi00ODNlLWE0MzktOTg3ZGE5NTBlZWFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI0VDE2MTMyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIzNTJjNjRjNDhhYmE3YjQ3Mzg3YjViMzg5YmE0NGI2YjBlNWFhMmRiYzJmNWM5NWNhNTZiYmNkNDhhY2U5N2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.J8iiomJYwJ1i2H6PIFuqCTKyXLRBI-aC4JSi0oA-LCU)

Ładnie napisane 🤔😉

> Over the weekend something changed on the npmjs.com site, and they have stopped supplying a documented field 'maintainers'.
> 
> * NPM always add maintainers to package, now stopped.
> * Homebridge had a bug that checked maintainers.lenght before checking if it exists at all. It was fixes. [Homebridge Wont Wake homebridge#2069](https://github.com/homebridge/homebridge/issues/2069)
> * Homebridge check maintainers from latest stable release of plugin, and first of maintainers you see on plugin card next to heart icon.
> * So if plugin doesnt have any new release since 2-3 days (since npm does not add maintainers), everything works.
> 
> To summarize:
> 
> * NPM error is that it does not add maintainers (since couple days).
> * The Homebridge UI error was that it checked how many maintainers, without checking if they were at all exist. And this suspended all npm functions, also responsible for checking the version, etc. If it weren't for that, the NPM error would only cause that there would be no author on the plugin card (only a heart icon and a @ sign because it is hardcoded).
> 
> To resolve:
> 
> * Manually update homebridge UI to the latest version 4.56.3
> 
> ![Screenshot 2024-06-24 at 09 05 42](https://private-user-images.githubusercontent.com/17454593/342236674-21f34241-c2b2-483e-a439-987da950eeab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkyNDYxMjEsIm5iZiI6MTcxOTI0NTgyMSwicGF0aCI6Ii8xNzQ1NDU5My8zNDIyMzY2NzQtMjFmMzQyNDEtYzJiMi00ODNlLWE0MzktOTg3ZGE5NTBlZWFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI0VDE2MTcwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1Y2RhZjYxYjMzZWY4MDdkZGM2MDRiMGM5ZmMxMzQ1N2E0YzEyMGRlYzY0NWQyOWM4MGM1NmMxYjc0MjJkYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4VYnAbvHUK62PDibFsv1ppr7slir7_IKfpJZBdKYsvs)

Nicely written 😉🤔 But.. #2070 not #2069

**Comment by anbie** [comment]
Works like a charm. All "Failed to check registry.npmjs.org" messages are gone. Thanks!

**Comment by mkz212** [comment]
> Over the weekend something changed on the npmjs.com site, and they have stopped supplying a documented field 'maintainers'.
> 
> * NPM always add maintainers to package, now stopped.
> * Homebridge had a bug that checked maintainers.lenght before checking if it exists at all. It was fixes. [Homebridge Wont Wake homebridge#2069](https://github.com/homebridge/homebridge/issues/2069)
> * Homebridge check maintainers from latest stable release of plugin, and first of maintainers you see on plugin card next to heart icon.
> * So if plugin doesnt have any new release since 2-3 days (since npm does not add maintainers), everything works.
> 
> To summarize:
> 
> * NPM error is that it does not add maintainers (since couple days).
> * The Homebridge UI error was that it checked how many maintainers, without checking if they were at all exist. And this suspended all npm functions, also responsible for checking the version, etc. If it weren't for that, the NPM error would only cause that there would be no author on the plugin card (only a heart icon and a @ sign because it is hardcoded).
> 
> To resolve:
> 
> * Manually update homebridge UI to the latest version 4.56.3
> 
> ![Screenshot 2024-06-24 at 09 05 42](https://private-user-images.githubusercontent.com/17454593/342236674-21f34241-c2b2-483e-a439-987da950eeab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkyNDYzMDIsIm5iZiI6MTcxOTI0NjAwMiwicGF0aCI6Ii8xNzQ1NDU5My8zNDIyMzY2NzQtMjFmMzQyNDEtYzJiMi00ODNlLWE0MzktOTg3ZGE5NTBlZWFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI0VDE2MjAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmYjc0Yzc1YTNjYzhhNmFmYzlkYjMxYzhmZGQ3ZWZmZDIzNTgzZWZmMDliOWRkOGFlZjgxNjkyYjJhYzk3OGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9uInJLjPFKKwpsFVgXglmw3rfsj7TQW8SPKcBqbJ9_k)

Nicely written 🤔😉 But #2070 , not #2069. And maybe write how to install other version (click on Homebridge UI in status widget...)

**Comment by dkerr64** [comment]
This should be closed as a duplicate of #2070

**Comment by mkz212** [comment]
I think you can already unpin this Issue.
